File: jasmine/jasmine-standalone-3/spec/bowlingSpec.js
describe('Bowling game', function() {

  var game;

  beforeEach(function () {
    game = new Bowling();
  });

  // it('bowls a ball', function() {
  //   expect(game.roll(5)).toEqual([5]);
  // });

  it('can roll a total score for 20 bowls', function() {
    for (var i = 0; i < 20; i++) {
      game.roll(2);
    }
    expect(game.score()).toEqual(40);
  });

  it('rolling a gutter score', function() {
    for (var i = 0; i < 20; i++) {
      game.roll(0);
    }
    expect(game.score()).toEqual(0);
  });

  it('scoring a strike', function() {
    game.roll(10);
    game.roll(2);
    game.roll(5);
    game.roll(4);
    game.roll(4);
    for (var i = 0; i < 15; i++) {
      game.roll(0);
    }
    expect(game.score()).toEqual(32);
  });

  it('rolling a spare', function () {
  game.roll(4);
  game.roll(6);
  game.roll(9);
  for (var i = 0; i < 17; i++) {
    game.roll(0);
  }
  expect(game.score()).toEqual(28);
  });

});

File: spec/bowling_spec.rb
require 'bowling'

describe 'Bowling Game Behaviour' do

  before do
    @game = BowlingGame.new
  end

  it 'can roll gutter game' do
    20.times{@game.roll(0)}
    expect(@game.score).to eq 0
  end

  it 'can roll all ones' do
    20.times{@game.roll(1)}
    expect(@game.score).to eq 20
  end

  it 'can roll a spare' do
    @game.roll(5)
    @game.roll(5)
    @game.roll(3)
    17.times{@game.roll(0)}
    expect(@game.score).to eq 16
  end

  it 'can roll a strike' do
    @game.roll(10)
    @game.roll(4)
    @game.roll(3)
    16.times{@game.roll(0)}
    expect(@game.score).to eq 24
  end

  it 'can rolla perfect game' do
    12.times{@game.roll(10)}
    expect(@game.score).to eq 300
  end

  it 'can roll all spares' do
    21.times{@game.roll(5)}
    expect(@game.score).to eq 150
  end
end
